test/alternator/test_returnvalues.py

```diff
@@ -441,3 +486,27 @@ def test_update_item_returnvalues_all_new_add_existing(test_table_s):
         UpdateExpression='ADD n :inc',
         ExpressionAttributeValues={':inc': 1})
     assert ret['Attributes']['n'] == 4
+
+# Testing feature which returns values only on modification condition failures.
+def test_update_item_returnvalues_on_condition_check_failure(test_table_s):
+    p = random_string()
+    test_table_s.put_item(Item={'p': p, 's': 'dog'})
+    # Succesfull modification doesn't return values.
+    ret=test_table_s.update_item(Key={'p': p},
+        ReturnValuesOnConditionCheckFailure='ALL_OLD',
+        ConditionExpression='s = :v1',
+        UpdateExpression='SET s = :v2',
+        ExpressionAttributeValues={':v1' : 'dog', ':v2': 'cat'})
+    assert not 'Attributes' in ret
+    # Failed condition request returns old values.
+    got_exception=False
+    try:
+        test_table_s.update_item(Key={'p': p},
+            ReturnValuesOnConditionCheckFailure='ALL_OLD',
+            ConditionExpression='s = :v1',
+            UpdateExpression='SET s = :v2',
+            ExpressionAttributeValues={':v1' : 'rat', ':v2': 'cow'})
+    except test_table_s.meta.client.exceptions.ConditionalCheckFailedException as err:
+        assert err.response['Item'] == {'p': {'S': p}, 's': {'S': 'cat'}}
+        got_exception=True
+    assert got_exception
```

Comment:
These tests are missing the following cases which we also need to verify (and as usual check it behaves like DynamoDB):
1. When you use ReturnValuesOnConditionCheckFailure of None, it works but the item is not returned on failure.
2. When you use ReturnValuesOnConditionCheckFailure of "ALL_NEW" or "dog", you get an error message.

Reply:
will add